src/engine.rs

```diff
@@ -842,19 +846,41 @@ impl<'a, 'b> Cpu<'a, 'b> {
                         self.hartid, self.state.pc, addr
                     );
                 }
-                // trace!("Load 0x{:x} ({}B)", addr, 8 << size);
-                self.engine
+                let word_offset = addr % 4;
+                let mask = (!(u64::MAX << (8 << size))) as u32;
+                if (size as u32) > (4 - word_offset) {
+                    warn!(
+                        "Hart {} (pc=0x{:08x}) is doing an unaligned load in DRAM at 0x{:08x}",
+                        self.hartid, self.state.pc, addr
+                    );
+                }
+                let shift = 8 * (word_offset);
+                let word = ((self
+                    .engine
                     .memory
                     .lock()
                     .unwrap()
-                    .get(&(addr as u64))
+                    .get(&((addr - word_offset) as u64))
                     .copied()
-                    .unwrap_or(0)
+                    .unwrap_or(0))
+                    >> shift)
+                    & mask;
+                debug!(
```

Comment:
Again, can you make this a `trace` to stay consistent with what we had before?

Reply:
Addressed in [this commit](https://github.com/pulp-platform/banshee/pull/18/commits/704b9b214fb4061a9d92881a5a2b382462fd08eb).